### DISCUSSION TITLE
[FR] Full page lock screen when another admin is editing an entry

### DISCUSSION THREAD

**Discussion by MadMikeyB** [craftcms/cms#15237] [Ideas] [answered]
Other CMS systems have a full page lock, with a modal that tells admin users when another admin user is actively editing an entry, WordPress example attached. Can this be made possible in Craft CMS, or a hook somehow so that Plugin developers can enable this?

![image](https://github.com/craftcms/cms/assets/831828/e4ab74a9-3d11-4366-b685-cb338aca5d77)

**Comment by brandonkelly** [comment, marked as answer]
Craft doesn’t do this natively because it’s generally safe for two people to be editing the same entry, so long as they’re working on different fields. Craft will show indicators when other authors are on the same entry, with tooltips explaining what they’re doing (#13420), and a notification will appear when the entry has been modified upstream, with a “Reload” button.

That said, I just [added a JavaScript event](https://github.com/craftcms/cms/commit/01fb01a3739255c6564ec05f014e92ef45a203b8) for Craft 5.3, which makes it possible to add this feature via a plugin or module:

```php
if (Craft::$app->request->getIsCpRequest() && !Craft::$app->request->getAcceptsJson()) {
    Craft::$app->view->registerJs(<<<JS
(() => {
  Garnish.on(Craft.ElementEditor, 'checkActivity', async (event) => {
    if (event.activity.some(record => record.type === 'edit')) {
      const elementEditor = event.target;
      const slideout = elementEditor.slideout;
      
      if (slideout) {
        await elementEditor.pause();
        slideout.$body.addClass('disabled').css('pointer-events', 'none');
        slideout.$noticeContainer.html('<p class="error">This entry is being edited by another user.</p>');
        slideout.$saveBtn.remove();
        return;
      }

      const backUrl = Craft.getUrl('entries');
      const backIcon = Craft.orientation === 'ltr' ? 'arrow-left' : 'arrow-right';
      const modalHtml = `
        <div class="modal fitted">
          <div class="body">
            <p>This entry is being edited by another user.</p>
            <a class="btn" href="\${backUrl}" data-icon="\${backIcon}">Go back</a>
          </div>
        </div>
      `;
      new Garnish.Modal(modalHtml, {
        hideOnEsc: false,
        hideOnShadeClick: false,
        shadeClass: 'modal-shade dark',
      });
      Garnish.off(Craft.ElementEditor, 'checkActivity', callback);
    }
  });
})();
JS);
}
```

With that code in place, edit pages will show an inescapable modal window alerting the user that another user is editing the same entry, the first time that it’s detected that someone is making changes to it.

<img src="https://github.com/craftcms/cms/assets/47792/4ab8c48a-8a79-44e0-ac7c-d4d39eefad70" width="638" height="432" alt="An example modal window stating that another user is editing the same entry, with a “Go back” button.">

If you want to test that out, you can update to the `5.3` branch by changing your `craftcms/cms` requirement in composer.json to `"5.3.x-dev as 5.3.0-alpha"`, and running `composer update`.

**Update:** the code example has been updated to also account for slideouts (requires Craft 5.7+).

**Reply by MadMikeyB** [reply]
This is perfect thank you @brandonkelly

**Comment by JaneConvergine** [comment]
Hi @brandonkelly thanks a lot for this! Your solution works perfectly.

Unfortunately, this does not work if users edit an entry with matrix blocks via a modal window/popup/sidebar (ajax loaded). 

Do you have a solution for that? 

I would greatly appreciate any help!

**Reply by brandonkelly** [reply]
Good point.

I just extended author indicator support to slideouts for Craft 5.7 (#16934). I’ve updated the code example above to account for slideouts as well.

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with that change.
